### PR TITLE
Remove final class from attachment uploader

### PR DESCRIPTION
**Pull request by integritty**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | After uploading files you need to do some actions with file, or do some modification. PrestaShop in version 1.7 / 8 doesn not contain any hook for this operation and it is not needed. You can do it using services.yml but only if core class is not final & `@internal`. This change allows you to override this class using module services.yml.
| Type?             | improvement
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Try to do symfony override from module using services.yml

### CODE REVIEW

**Comment by integritty** [comment]
This pull request following slack message
Screenshot 2022-05-18 at 9 20 46

**Review by Progi1984 — CHANGES_REQUESTED** [review]
@integritty Could you change `private` methods & properties to `protected` ?

**Comment by integritty** [comment]
@Progi1984 Done.

**Review by Progi1984 — COMMENTED** [review]
@integritty Don't forget properties :wink:

**Comment by integritty** [comment]
Its ok now?

**Review by Progi1984 — DISMISSED** [review]

File: src/Adapter/File/Uploader/AttachmentFileUploader.php [hunk 1]
```diff
@@ -75,7 +80,7 @@ public function upload(
         string $uniqueFileName,
         int $fileSize,
         int $id = null,
-        $throwExceptionOnFailure = true
+        bool $throwExceptionOnFailure = true
```
Comment (MeKeyCool) [inline src/Adapter/File/Uploader/AttachmentFileUploader.php:82]: :+1:

**Review by MeKeyCool — DISMISSED** [review]
Thanks for your contribution ^^

For the css-fixer CI problem, you may look for this command : `php ./vendor/bin/php-cs-fixer fix`

**Review by Progi1984 — DISMISSED** [review]

**Review by MeKeyCool — DISMISSED** [review]
You still have an error in CI :thinking: It looks linked to your PR comment.
Did you remove first comment ?
```
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/8/contribute/contribution-guidelines/#pull-requests
------------------------------------------------------------------------------>
```
?

**Comment by integritty** [comment]
> You still have an error in CI 🤔 It looks linked to your PR comment. Did you remove first comment ?
> 
> ```
> 
> ```
> 
> ?

I tried to remove this comment lines and force push the same changes

**Review by MeKeyCool — DISMISSED** [review]

**Comment by matks** [comment]
I close and reopen the PR to restart the CI

**Review by matks — DISMISSED** [review]
Change is valid for me

**Comment by integritty** [comment]
@matks Is everything ok with this pull request? Anything else you want me to do?

**Comment by NeOMakinG** [comment]
@integritty I don't get why the CI is 🔴 , could you maybe try to rebase it?

**Review by kpodemski — DISMISSED** [review]

**Comment by integritty** [comment]
@NeOMakinG Its ok now?

**Review by FabienPapet — APPROVED** [review]

**Review by NeOMakinG — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Comment by kpodemski** [comment]
Thank you, @integritty

**Pull request merged**